public/pages/CreateMonitor/containers/DefineMonitor/DefineMonitor.js

```diff
@@ -168,24 +171,36 @@ class DefineMonitor extends Component {
     const { httpClient, values } = this.props;
     const formikSnapshot = _.cloneDeep(values);
 
-    // If we are running a visual graph query, then we need to run two separate queries
-    // 1. The actual query that will be saved on the monitor, to get accurate query performance stats
-    // 2. The UI generated query that gets [BUCKET_COUNT] times the aggregated buckets to show past history of query
-    // If the query is an extraction query, we can use the same query for results and query performance
-    const searchRequests = [buildSearchRequest(values)];
-    if (values.searchType === SEARCH_TYPE.GRAPH) {
-      searchRequests.push(buildSearchRequest(values, false));
+    const searchType = values.searchType;
+    let generalRequests;
+    switch (searchType) {
+      case SEARCH_TYPE.QUERY:
+        generalRequests = [buildSearchRequest(values)];
+        break;
+      case SEARCH_TYPE.GRAPH:
+        // If we are running a visual graph query, then we need to run two separate queries
+        // 1. The actual query that will be saved on the monitor, to get accurate query performance stats
+        // 2. The UI generated query that gets [BUCKET_COUNT] times the aggregated buckets to show past history of query
+        // If the query is an extraction query, we can use the same query for results and query performance
+        generalRequests = [buildSearchRequest(values)];
+        generalRequests.push(buildSearchRequest(values, false));
+        break;
+      case SEARCH_TYPE.HTTP:
+        generalRequests = [buildHTTPRequest(values)];
+        break;
     }
 
     try {
-      const promises = searchRequests.map(searchRequest => {
+      const promises = generalRequests.map((generalRequest) => {
```

Comment:
Rename the variable to make this `request` take care of both "searchRequest" and "httpRequest", not sure if it's a good name.